tools/hive-bigquery/hive_to_bigquery.py

```diff
@@ -0,0 +1,241 @@
+
+    Args:
+        mysql_component (:class:`MySQLComponent`): Instance of MySQLComponent
+        to connect to MySQL
+        hive_table_model (:class:`HiveTableModel`): Wrapper to Hive table
+            details
+    """
+
+    print_and_log("Rolling back...", logging.INFO)
+    mysql_component.drop_table_if_empty(hive_table_model.tracking_table_name)
+    print_and_log("Rollback success", logging.INFO)
+    exit()
+
+
+def main():
+    """Migrates Hive tables to BigQuery
+
+    Establishes connection to Hive, MySQL, GCS and BigQuery. Validates the
+    user arguments and continues migration from the previous runs, if any.
+    """
+    try:
+        init_script.initialize_variables()
+
+        logger.debug("Initializing Properties Reader")
+        PropertiesReader('application.properties')
+
+        # Initializes the components to connect to MySQL, GCS, BigQuery and Hive
+        mysql_component = MySQLComponent(
+            PropertiesReader.get('tracking_database_host'),
+            PropertiesReader.get('tracking_database_user'),
+            PropertiesReader.get('tracking_database_password'),
+            PropertiesReader.get('tracking_database_db_name'),
+            PropertiesReader.get('tracking_database_port'),
+        )
+        gcs_component = GCSStorageComponent(PropertiesReader.get('project_id'))
+        bq_component = BigQueryComponent(PropertiesReader.get('project_id'))
+        hive_component = HiveComponent(
+            PropertiesReader.get('hive_server_host'),
+            PropertiesReader.get('hive_server_port'),
+            PropertiesReader.get('hive_server_username')
+        )
+
+        # Validates the user provided resources
+        logger.debug("Validating the resources")
+        if ResourceValidator.validate(hive_component, gcs_component,
+                                      bq_component):
+            logger.info("All the provided resources are valid")
+        else:
+            print_and_log("Check the provided resources", logging.CRITICAL)
+            exit()
+
+        hive_table_object = HiveTable(
+            hive_component, PropertiesReader.get('hive_database'),
+            PropertiesReader.get('hive_table_name'),
+            PropertiesReader.get('incremental_col')
+        )
+        # Wrapper to describe Hive table resource
+        hive_table_model = hive_table_object.hive_table_model
+        logger.debug(hive_table_model)
+
+        bq_table_object = BigQueryTable(PropertiesReader.get('dataset_id'),
+                                        PropertiesReader.get('bq_table'),
+                                        hive_table_model)
+        # Wrapper to describe BigQuery table resource
+        bq_table_model = bq_table_object.bq_table_model
+        logger.debug(bq_table_model)
+
+    except Exception as error:
+        logger.exception(error)
+        print_and_log("Check the log file for detailed errors",
+                      logging.CRITICAL)
+        exit()
+
+    try:

```

Comment:
Try block is very large and has many unrelated statements. Can we make try block specific and move the ones which are executed subsequently upon success to the else block of the try except else structure